tests/unit/setup-rendering-context-test.js

```diff
@@ -555,6 +561,295 @@ module('setupRenderingContext', function (hooks) {
+          const name = 'Chris';
+
+          const template = precompileTemplate(
+            '<p>hello my name is {{name}}</p>',
+            {
+              scope() {
+                return {
+                  name,
+                };
+              },
+            }
+          );
+          const component = setComponentTemplate(template, templateOnly());
+          await render(component);
+          assert.equal(
+            this.element.textContent,
+            'hello my name is Chris',
+            'has rendered content'
+          );
+        });
+
+        test('works with a glimmer component', async function (assert) {
+          const name = 'Chris';
+          const template = precompileTemplate(
+            '<p>hello my name is {{name}} and my favorite color is {{this.favoriteColor}}</p>',
+            {
+              scope() {
+                return {
+                  name,
+                };
+              },
+            }
+          );
+
+          class Foo extends GlimmerComponent {
+            favoriteColor = 'red';
+          }
+
+          setComponentTemplate(template, Foo);
+          await render(Foo);
+
+          assert.equal(
+            this.element.textContent,
+            'hello my name is Chris and my favorite color is red',
+            'has rendered content'
+          );
+        });
+
+        test('works with a classic component', async function (assert) {
+          const name = 'Chris';
+          const template = precompileTemplate(
+            '<p>hello my name is {{name}} and my favorite color is {{this.favoriteColor}}</p>',
+            {
+              scope() {
+                return {
+                  name,
+                };
+              },
+            }
+          );
+
+          const Foo = Component.extend({
+            favoriteColor: 'red',
+          });
+
+          setComponentTemplate(template, Foo);
+          await render(Foo);
+
+          assert.equal(
+            this.element.textContent,
+            'hello my name is Chris and my favorite color is red',
+            'has rendered content'
+          );
+        });
+
+        test('components passed to render do *not* have access to the test context', async function (assert) {
+          const template = precompileTemplate(
+            'the value of foo is {{this.foo}}'
+          );
+
+          class Foo extends GlimmerComponent {
+            foo = 'foo';
+          }
+
+          const component = setComponentTemplate(template, Foo);
+          await render(component);
+
+          assert.equal(this.element.textContent, 'the value of foo is foo');
+
+          this.set('foo', 'bar');
+          await rerender();
+          assert.equal(this.element.textContent, 'the value of foo is foo');
+        });
+
+        test('setting properties on the test context when rendering a component triggers a warning', async function (assert) {
```

Comment:
Let's tweak this to be:

* an assertion
* move the `set`'s down after `await render` (which I think is what most folks probably do)